aten/src/ATen/native/cpu/vec256.h

```diff
@@ -0,0 +1,165 @@
+#pragma once
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+#include <x86intrin.h>
+
+constexpr size_t _THRESHOLD = 32768;
```

Comment:
I think this doesn't belong to the `vec` header. It's an implementation detail of `tbb` and it would be different if we used e.g. OpenMP

Reply:
I think in general it's worth splitting out the parallelism patterns into a separate header for 1d and 2d reductions. Then we can also be more flexible on the backend etc. and know our specifications.